### PR TITLE
Fill contiguity according to stride_order. 

### PR DESCRIPTION
**Pull request by wujingyue**
This will fix a bug introduced by my https://github.com/NVIDIA/Fuser/pull/3145.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -1118,13 +1118,20 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+            const auto rank = static_cast<int64_t>(shape.size());
+            std::vector<std::optional<bool>> contiguity_vec(rank);
+            // This duplicates some code around
+            // https://github.com/NVIDIA/Fuser/blob/b60f2341bbe2ec276b1fe60f4f25a4a5b093faa9/csrc/python_frontend/fusion_record.h#L1370.
+            // Alternatively, I can extend TensorRecord to allow contiguity as
+            // a boolean. If you think it's worth doing, I'm happy to pursue it
+            // in a separate PR.
+            for (const auto index : c10::irange(rank)) {
+              const auto contig_index =
+                  stride_order.empty() ? index : rank - 1 - stride_order[index];
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:1130]: ah... I totally forgot about this one. :face_with_head_bandage:

File: tests/python/test_define.py [hunk 2]
```diff
@@ -5,36 +5,50 @@
...
-    inp_tensor = torch.randn(1, 2, 1, device="cuda").as_strided([1, 2, 1], [0, 1, 0])
-    out_tensor = fd.execute([inp_tensor])[0]
-    torch.testing.assert_close(out_tensor, inp_tensor * 2)
+from utils import NVFuserTest
+
+
+# I tried to merge the tests to opinfo and
+# [failed](https://github.com/NVIDIA/Fuser/commit/b0ccb481a7d60f944abf4b6b164f625fec31d147).
+# There seems to be a bug in opinfo causing the generated test to pass fewer
+# arguments than parameters.
```
Comment (jjsjann123) [inline tests/python/test_define.py:14]: hmmm... that sounds strange. Thanks for logging it here. I wonder if we should have this as an issue as well?

File: tests/python/test_define.py [hunk 3]
```diff
@@ -5,36 +5,50 @@
...
+            out = fd.ops.add(inp, inp)
+            fd.add_output(out)
+
+        in_tensor = torch.randn(1, 2, 1, device="cuda").as_strided([1, 2, 1], [0, 1, 0])
+        out_tensors, _ = self.exec_nvfuser(fusion_func, [in_tensor])
+        torch.testing.assert_close(out_tensors[0], in_tensor * 2)
+
+    def test_contiguity_with_stride_order(self):
+        def fusion_func(fd: FusionDefinition):
+            inp = fd.define_tensor([1, 2], contiguity=True, stride_order=[0, 1])
```
Comment (jjsjann123) [inline tests/python/test_define.py:48]: :clap:

**Review by jjsjann123 — APPROVED** [review]

File: tests/python/test_define.py (hunk 2 above)
Comment (wujingyue) [reply tests/python/test_define.py]: Done!

**Comment by wujingyue** [comment]
!build

**Pull request merged**